### DISCUSSION TITLE
Is My UVtools Update History Stored Anywhere on My Installation?

### DISCUSSION THREAD

**Discussion by Polydyne** [sn4k3/UVtools#631] [Q&A] [answered]
Thank you for this great tool. It's an incredible effort on your part, and has solved countless print problems for us.

My question is this: Is a update/revision installation history for UVtools stored anywhere on my machine? I am attempting to revert back to an earlier version that was producing perfect prints.

Background: began using UVtools about a year ago, and made several version updates (but don't think I made every one offered). After a recent update within the last month or two, UVtools began altering print parameters from the AnyCubic .pwms files being analyzed/edited. If edited file is then saved from UVtools, printer runs that file with corrupted print parameters. For example, original AnyCubic file will have exposure at 3.50 sec and retract speed at 40.5 mm/min. After passing thru UVtools, exposure will be set at 350 secs and retract speed at 405 mm/min (or sometimes less obvious errors i.e. 40.5 mm/min becomes 540 mm/min). Prints immediately fail.

Appears to be parsing errors in file import from AnyCubic Photon WorkShop into UVtools, but our best path back to good prints is to undo our UVtools version updates back in time until we reach the version that ran so well.

Typ. process path: CAD model export > AnyCubic Photon Workshop supports/slicing > UVtools issue analysis/pixel edits > AnyCubic Mono SE print from UVtools 'zero-issues' edited slice file

Thanks for any help!

**Comment by sn4k3** [comment, marked as answer]
UVtools does not keep history of that. If you are using MSI maybe Windows keep track of it on Events, but I don't know.

> For example, original AnyCubic file will have exposure at 3.50 sec and retract speed at 40.5 mm/min. After passing thru UVtools, exposure will be set at 350 secs and retract speed at 405 mm/min (or sometimes less obvious errors i.e. 40.5 mm/min becomes 540 mm/min). Prints immediately fail.

That's strange, it's multiplying by 100...
Can you provide two files? One original and the same modified by UVtools?

**Reply by Polydyne** [reply]
Hello and thank you for reply.

I will provide two files, orig & edited w/corruption, but will have to do some digging. Will post them here in next day or so. Thanks for patience, as we have pile of files generated while attempting diagnosis & fix here (Also new to Github, so a bit slow as we become familiar)

For the record, not always multiplyX100 corruption, just used that as an example. Actually think I saw more transposed digits i,e, 40.5 sec becomes 504 sec. Also think I had couple of random corruptions...40.5 becomes 492048297. Single digit parameters seem to hold...6 bottom layers imports & remain as 6 correctly.

Also not all imported parameters corrupted, but did see consistently corrupted bottom exposure, regular exposure, lift speed, retract speed, maybe some others (and maybe support & raft geometries corrupted,,,not yet confirmed for certain)

Will get you two files as requested A.S.A.P.

And thank you again for reply, help & amazing UVtools!!!

**Reply by sn4k3** [reply]
From code perspective is not doing any math over that fields.
What is your PC system?

**Reply by Polydyne** [reply]
HP Z820 workstation 64GB ram Win10 Pro 64

Have been working on this all day. Need to make some corrections to my earlier comments.

Parameter corruption I  described before may not be happening. PhotonWorkshop speeds are in mm per sec, UVtools speeds shown in mm/min, so 4.00mm/sec Photon becomes 240mm/min UVt, which is correct. 
>>> so when prints went bad, I may have erred and adjusted UVt number 240 down closer to Photon number 40 making prints even worse.

I'm starting to think problems came from PhotonWorkshop update changing default parameters, because files that printed perfect 2 weeks ago were suddenly trash. Know for a fact that exposure times, support & raft geometries, and speeds were changed.

To complicate things, PhotonWs can't load it's own sliced files for edit/review. So slice in Photon > import to UVt & edit & save. Then only way to further review/edit is in UVt. No way to find original PhotonWs parameters or support geometries.

At this point, I will continue trying to get back to known good prints, but please don't waste your time looking for UVt code bugs that may not exist. PhotonWs is clunky & buggy and I should have NEVER updated it...if it works don't fix it.

For what its worth, I created two sample files below from PhotonWs & (PhotonWs +UVtools analysis) 
1. CAD>PhotonWs>Support>Slice>save as PhW
2. PhW>UVtools>analyze>zero issues>save as UVt

I will attempt to print both under the assumption that printer will make 2 good parts, 2 bad parts, or 1 good/1bad

Thanks again for your help!

P.S. re original question: My Win history only showed most recent update to UVtools. But I found your complete update Rev history on GitHub and if necessary I can step back thru revs one-by-one.

[Debug_Clamp_29_01_UVt.zip](https://github.com/sn4k3/UVtools/files/10297889/Debug_Clamp_29_01_UVt.zip)
[Debug_Clamp_29_01_PhW.zip](https://github.com/sn4k3/UVtools/files/10297890/Debug_Clamp_29_01_PhW.zip)

**Reply by sn4k3** [reply]
> To complicate things, PhotonWs can't load it's own sliced files for edit/review.

They allow you to open the sliced file and review parameters, both global and per layer.

> I'm starting to think problems came from PhotonWorkshop

Why use PW anyway? Even chitubox is better, you also have lychee and prusaslicer.

> For what its worth, I created two sample files below from PhotonWs & (PhotonWs +UVtools analysis)

I can confirm that both files have the same parameters, nothing abnormal.

As advice I would suggest to not use AntiAliasing on anycubic machines until they proper fix it. Some models are known to trash your print

**Reply by Polydyne** [reply]
>They allow you to open the sliced file and review parameters, both global and per layer.

You are correct, I misspoke again. Should have said PW can't re-load it's sliced file for placement/support edits, so if UVt finds big issues, have to start from scratch CAD>PW>placement/supports>slice>UVt analysis. (at least with my old version of PW and my knowledge of using it)

>Why use PW anyway? Even chitubox is better, you also have lychee and prusaslicer.

We tend to chase fastest path to functional parts which sometimes costs us time later. AC Photon MonoSE shipped with PW, so we used it & got it working. At one point we installed Lychee, but didn't commit to learning it as we just wanted parts. Probably should revisit now.

>I can confirm that both files have the same parameters, nothing abnormal.

Thank you for checking. Gets us thru fog of not know where problems are originating.

>As advice I would suggest to not use AntiAliasing on anycubic machines until they proper fix it. Some models are known to trash your print

Thanks for that valuable input. We did not know that, and so again lost in the fog of not knowing root problems. (FWIW we did not intentionally set A-A, which raises question of whether it was set by accident or changed by software update.)

This entire platform is new to our workflow (12months), with all parts for precision functional proto assemblies. So the fact that we even reached that point should make us happy. Now if we can get repeatability, we're golden. (and if we can get the guy making decisions to stop changing resins on us) That guy would be me. LOL! 

Thanks for all the help!
Have a great day & Happy Holidays!

**Reply by sn4k3** [reply]
>  PW can't re-load it's sliced file for placement/support edits

Few slicers can do that but if you do you get a bad quality print, you should NEVER open and edit a sliced file. Instead you should save the project and reslice, or do not close slicer inspect with UVtools and live update on slicer.

> We tend to chase fastest path to functional parts which sometimes costs us time later. AC Photon MonoSE shipped with PW, so we used it & got it working. At one point we installed Lychee, but didn't commit to learning it as we just wanted parts. Probably should revisit now.

If you want to save time don't use PW. Go with Lychee, PrusaSlicer or Chitubox. Tango is also very good but paied.
As future not avoid Anycubic machines until they make proper firmwares

**Reply by Polydyne** [reply]
UPDATE:
Got back to good parts. Several problems found, and we learned a lot (but it wasn't fun)
1. We had stepwise pushed print speed up with motion & exposure times. We didn't realize we were close to going from perfect to junk.
2. Cold weather snap pushed the process out of preferred temp range of resin. We added some heat to the room.
3. New resin won't run with old resin settings. Whole different animal.
4. Found one bad print parameter that we think came preset on a PW update. A thickness or height that was supposed to be 2.0mm but was set to 0.02mm.
5. As mentioned earlier, we missed that speeds in PW are mm/sec and in UVt are mm/min. Our blunder was to change speeds within UVt to numbers we remembered from PW.
6. We shrunk support counts & dimensions based on old resin. New resin did not like that at all.

All in all, we ended up spending a few days reviewing old print files, old app version presets, etc and then reset every print parameter to known good numbers. Upped the room temp, added more & beefier supports, and sliced fresh. Ran UVt, got zero issues, and printed perfect (but slow) parts.

Hope some of this is helpful to others, and thanks again for your help sn4k3 in sorting thru what was a real mess.

And remember: if it ain't broke, DON'T FIX IT!  LOL

P.S. UVtools ROCKS!

**Reply by sn4k3** [reply]
1. People are never happy, they always want faster and faster, crazy methods surge on internet to degrade quality over speed. If you have to reprint you wont save time, instead you lose! Better slow and perfect than fast with failrate.
2. PTC Heater inside printer is what people often do.
3. New resin always need calibration, even inside same brand/model.

> And remember: if it ain't broke, DON'T FIX IT! LOL

+1

> P.S. UVtools ROCKS!

For sure ;)
